packages/core/src/modules/oob/OutOfBandApi.ts

```diff
@@ -565,6 +564,20 @@ export class OutOfBandApi {
    * @param outOfBandId the out of band record id
    */
   public async deleteById(outOfBandId: string) {
+    const outOfBandRecord = await this.getById(outOfBandId)
+
+    const relatedConnections = await this.connectionsApi.findAllByOutOfBandId(outOfBandId)
```

Comment:
This can become quite an expensive query if you have a lot of connections created. Another reason to support pagination...

Reply:
Yes... Hopefully, delete operations will not be so common, so the performance impact would not be so big.